labs/a8.md

```diff
@@ -0,0 +1,745 @@
+would change the file's owner and owning group, previously "staff" and "staff", 
+to you. The basic syntax of `chown` is fairly simple:
+
+    chown [-R (recurse)] [user]:[group] PATH
+    
+where `PATH` is the file or directory whose ownership you wish to modify.
+
+Making sure that files are only accessible to those who should be allowed to do 
+so, and that vulnerable programs are not given too many permissions, is a 
+critical part of maintaining system security on Linux.
+
+#### Task 6 - File Security
+
+Lets practice using the commands and concepts covered above.
+
+`cd` into the folder `/opt/lab8` on your student VM. You should see a number of 
+files named `file1.txt` through `file9.txt`. Please answer the following 
+questions with regards to each of the files, and make note of what commands you 
+use for checkoff.
+
+1. `file1.txt`: make this file readable using `chmod` and answer the question 
+contained therein.
+2. `file2.txt`: if you read this file, you'll notice it contains important 
+decal secrets. Use `chmod` and `chown` to make the file readable only to you. 
+3. `file3.txt`: this is an even more important file. Only root should be able 
+to read this file, and no one should be able to edit it.
+4. `file4.txt` and `file5.txt`: these files are owned by another user. Choose 
+any method to make the files readable to you and unreadable to the previous 
+owner.
+5. `file6.txt` and `file7.txt`: look at the permissions for these two files and 
+explain what they allow and disallow.
+6. `file8.txt` and `file9.txt`: provide a strategy to make these files readable 
+only to you and the lab8user, and no one else. 
+
+#### Network Security 
+
+On UNIX based operating systems, the network is the most common method of 
+gaining unauthorized access to a machine. Why this is the case should be 
+obvious: the network allows one to interact with a machine without needing any 
+physical presence. Anyone on the same network as your machine can connect, and 
+therefore attack, your machine. On the internet, this means the whole world can 
+attack your machine. This introduces a number of security considerations when 
+running networked applications.
+
+The first step in securing machines connected to a network, or the open 
+internet, is to make it as hard as possible for attackers to get to your 
+machine. As you may recall from the networking lecture, processes interact with 
+one another over a network by means of sockets and ports. Many protocols listen 
+on well-known ports, such as SSH on port 22, HTTP on port 80, FTP on port 21, 
+etc. We use firewalls and similar software to prevent unauthorized users from 
+connecting to your machine. For example, the student VMs are heavily firewalled 
+- the only traffic allowed from the internet at large is SSH (TCP to port 22), 
+HTTP/S (TCP to port 80, 443, 8000, 8080) and a few other services necessary for 
+labs (e.g. UDP to port 53 for lab 7).
```

Comment:
This isn't true on advanced VMs, right? It was last semester, but it is not now.